lib/src/where_type.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+/// Emits only the events which have type [R].
+///
+/// If the source stream is a broadcast stream the result will be as well.
+///
+/// Errors from the source stream are forwarded directly to the result stream.
+StreamTransformer<Null, R> whereType<R>() => _WhereType<R>();
+
+class _WhereType<R> extends StreamTransformerBase<Null, R> {
+  @override
+  Stream<R> bind(Stream<Object> values) {
```

Comment:
This matches the naming pattern used elsewhere in the package.
https://github.com/dart-lang/stream_transform/blob/5c5049333dcef448acebb09c6d53b6c99c1b99e8/lib/src/from_handlers.dart#L50

Reply:
Do you think its actually a good name though?

Reply:
I think it's about as good as anything else. A `value` or `event` is what is emitted by a stream when we don't know anything more specific about it, and I generally like to name a stream as the plural of whatever flows through it. I don't like the name `inputs` because I wouldn't call the thing emitted by the stream an `input` - I wouldn't name the argument to the listen callback that.

Reply:
> I don't like the name `inputs` because I wouldn't call the thing emitted by the stream an `input`

In the case of a stream transformer I would disagree - it is transforming one stream into another stream, so it makes sense to me to consider one the "inputs" stream and one the "outputs" stream.

If I see the word "values" in the middle of the method its not clear what that represents. Is that the outputs or the inputs? But if I see "inputs" I can immediately assume its the input stream.

In any case I don't care enough to withhold an lgtm, but I do think "inputs" is better :).

> I wouldn't name the argument to the listen callback that.

Right, because listen doesn't transform anything, it only listens for values and there is only one stream in scope. With a transformer there are two streams in the scope of the `bind` method.